rotkehlchen/chain/evm/decoding/cctp/decoder.py

```diff
@@ -0,0 +1,129 @@
+            token_decimals=6,  # USDC has 6 decimals
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset.identifier == self.asset_identifier and
+                event.balance.amount == deposit_amount and
+                event.location_label == user_address
+            ):
+                chain_info = ''
+                try:
+                    chain_info = f' from {self.evm_inquirer.chain_id.label()} to {CCTP_DOMAIN_MAPPING[to_chain].label()}'  # noqa: E501
+                except KeyError:
+                    log.error(f'Could not find chain ID {to_chain} for CCTP transfer from {self.evm_inquirer.chain_name}')  # noqa: E501
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = f'Bridge {event.balance.amount} USDC{chain_info} via CCTP'
+                event.counterparty = CPT_CCTP
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        if not self.base.is_tracked(user_address := hex_or_bytes_to_address(context.tx_log.topics[1])):  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        deposit_amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[0:32]),
+            token_decimals=6,  # USDC has 6 decimals
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset.identifier == self.asset_identifier and
+                event.balance.amount == deposit_amount and
+                event.location_label == user_address
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = f'Bridge {event.balance.amount} USDC via CCTP'
+                event.counterparty = CPT_CCTP
+
+                # add information about the chain the USDC was bridged from
+                for log_event in context.all_logs:
+                    if log_event.topics[0] == MESSAGE_RECEIVED:
+                        from_chain = hex_or_bytes_to_int(log_event.data[:32])
+                        try:
+                            chain_info = f' from {CCTP_DOMAIN_MAPPING[from_chain].label()} to {self.evm_inquirer.chain_id.label()}'  # noqa: E501
+                            event.notes = f'Bridge {event.balance.amount} USDC{chain_info} via CCTP'  # noqa: E501
+                        except KeyError:
+                            log.error(f'Could not find chain ID {from_chain} for CCTP transfer to {self.evm_inquirer.chain_name}')  # noqa: E501
```

Comment:
no need to iterate the logs. The decoding logic already iterates the logs. You can add a rule for the address that emits `MESSAGE_RECEIVED` and then loop trough the decoded events loking for withdrawal/bridge/cpt_cctp and editing the notes.